python_modules/dagster/dagster/core/definitions/time_window_partitions.py

```diff
@@ -140,8 +140,8 @@ def time_window_for_partition_key(self, partition_key: str) -> TimeWindow:
             ),
             execution_timezone=self.timezone,
         )
-        next(iterator)
-        return TimeWindow(start, next(iterator))
+
+        return TimeWindow(next(iterator), next(iterator))
```

Comment:
With the introduction of minute, hour, and day offsets, an issue in this function became apparent. Since the hour/minute offset is truncated from the partition key, returning `TimeWindow(start, next(iterator))` would return an incorrect window. 

For example, if I had daily partition with offset of 4:15, then my partition key is still `2022-03-28` and that means that `2022-03-28-00:00` would be set as the beginning of the returned time window.

Returning TimeWindow(next(iterator), next(iterator)) instead will return the first partition starting on or after `partition_key`.